sw/applications/spi_host_dma_example/spi_host_dma_example.c

```diff
@@ -68,28 +69,24 @@ int main(int argc, char *argv[])
     // SPI and SPI_HOST are the same IP so same register map
     uint32_t *fifo_ptr = spi_host.base_addr.base + SPI_HOST_DATA_REG_OFFSET;
 
-    // DMA CONFIGURATION --
-    // dma peripheral structure to access the registers
-    dma_t dma;
-    dma.base_addr = mmio_region_from_addr((uintptr_t)DMA_START_ADDRESS);
+    // -- DMA CONFIGURATION --
     dma_set_read_ptr_inc(&dma, (uint32_t) 0); // Do not increment address when reading from the SPI (Pop from FIFO)
+    dma_set_write_ptr_inc(&dma, (uint32_t) 4); // Do not increment address when reading from the SPI (Pop from FIFO)
     dma_set_read_ptr(&dma, (uint32_t) fifo_ptr); // SPI RX FIFO addr
     dma_set_write_ptr(&dma, (uint32_t) copy_data); // copy data address
-
     // Set the correct SPI-DMA mode:
     // (0) disable
     // (1) receive from SPI (use SPI_START_ADDRESS for spi_host pointer)
     // (2) send to SPI (use SPI_START_ADDRESS for spi_host pointer)
     // (3) receive from SPI FLASH (use SPI_FLASH_START_ADDRESS for spi_host pointer)
     // (4) send to SPI FLASH (use SPI_FLASH_START_ADDRESS for spi_host pointer)
     #ifndef USE_SPI_FLASH
-        dma_set_spi_mode(&dma, (uint32_t) 1); // The DMA will wait for the SPI rx FIFO valid signal
+        dma_set_spi_mode(&dma, (uint32_t) 1); // The DMA will wait for the SPI RX FIFO valid signal
     #else
-        dma_set_spi_mode(&dma, (uint32_t) 3); // The DMA will wait for the SPI rx FIFO valid signal
+        dma_set_spi_mode(&dma, (uint32_t) 3); // The DMA will wait for the SPI FLASH RX FIFO valid signal
     #endif
 
-    // ---------------------
-
+    // -- SPI CONFIGURATION -- 
     // Configure chip 0 (flash memory)
     // Max 50 MHz core SPI clock --> Max 25 MHz SCK
     // Single data IO; keep timing as safe as possible.
```

Comment:
Remove this and check if there is any limitation on the max SPI clock for the Flash.